### PR TITLE
Feature/moving to mongodb

### PR DESCRIPTION
**Pull request by nblogist**
Moving the backend from NeDB to MongoDB using monk.

### CODE REVIEW

**Review by carlhong — APPROVED** [review]

File: src/crawler/getPolkadotData.ts [hunk 1]
```diff
@@ -38,7 +38,7 @@ export async function runCrawlers(provider, types, storage) {
...
+                    payload.call_data = _datumPair.extrinsic.method.method == 'asMulti' ? _datumPair.extrinsic.method.args[3].toHuman()?.toString()! : (await storage.find({ "call_hash": _datumPair.extrinsic.method.args[3].toHuman()?.toString()! }) as Array<any>)[0]?.call_data; // We don't get call_data until the final approval call (as_multi), so I think the call_data for other calls would be empty. but in case we had a call_data in DB before from some other final call of as_multi then we'd have its data and we'd add it here.
 
                     payload.status = event.method == 'NewMultisig' && (_datumPair.extrinsic.method.method == 'approveAsMulti' || _datumPair.extrinsic.method.method == 'asMulti') ? 'created' : event.method == 'MultisigApproval' && _datumPair.extrinsic.method.method == 'approveAsMulti' ? 'approving' : event.method == 'MultisigExecuted' && _datumPair.extrinsic.method.method == 'asMulti' ? 'executed' : event.method == 'MultisigCancelled' && _datumPair.extrinsic.method.method == 'cancelAsMulti' ? 'cancelled' : '';
```
Comment (wuminzhe) [inline src/crawler/getPolkadotData.ts]: I don't think you need to check the extrinsic name.
And a line should not be too long, it's hard to read. maybe use if or switch here.

**Comment by wuminzhe** [comment]
maybe you can query the database first, and then modify the data by using event and extrinsic. 
```ruby
if NewMultisig(approving, multisig, call_hash) 
    id = "#{block_number}_#{event_index}"
    if not_exist_in_db(id, call_hash)
      create a new call in db
    end
end

if MultisigApproval(approving, timepoint, multisig, call_hash)
  id =  "#{timepoint. height}_#{timepoint.index}"
  call = find_in_db(id, call_hash)
  call.approvals << approving
  update to db
end

...
```

File: src/crawler/getPolkadotData.ts (hunk 1 above)
Comment (nblogist) [reply src/crawler/getPolkadotData.ts]: I agree, just the event name will do too.
ahh, just realized line wrapping was on in my editor, didn't realize the line length reached 514. :grin:

**Comment by nblogist** [comment]
In one way this is implemented in the project, 

the only difference is that it makes the payload and then check if it is present or not, and in case of approvals it is querying the DB first, like you suggested.
But I agree that matching the current timepoint, hieght and index and blocknumber and event index only is minimalist but  refactoring the current version to accommodate for that won't bring much of a performance difference and might leave code prone to bugs.

**Comment by nblogist** [comment]
merged in error

**Comment by nblogist** [comment]
Moved the PR here: https://github.com/itering/subscan-multisig-backend/pull/6

**Pull request merged**